game/lib/journeys/journeys_page.dart

```diff
@@ -132,18 +144,95 @@ class _JourneysPageState extends State<JourneysPage> {
                 ),
               ),
             ),
-            Expanded(
-              child: ListView.separated(
+            Expanded(child: Consumer5<EventModel, GroupModel, TrackerModel,
+                    ChallengeModel, UserModel>(
+                builder: (context, myEventModel, groupModel, trackerModel,
+                    challengeModel, userModel, child) {
+              List<Widget> eventCells = [];
+              if (myEventModel.searchResults == null) {
+                myEventModel.searchEvents(
+                    0,
+                    1000,
+                    [
+                      EventRewardType.PERPETUAL,
+                      EventRewardType.LIMITED_TIME_EVENT
+                    ],
+                    false,
+                    false,
+                    false);
+              }
+              print("search results: " + myEventModel.searchResults.toString());
+              final events = myEventModel.searchResults ?? [];
+              if (!events
+                  .any((element) => element.id == groupModel.curEventId)) {
+                final curEvent =
+                    myEventModel.getEventById(groupModel.curEventId ?? "");
+                if (curEvent != null) events.add(curEvent);
+              }
+              for (EventDto event in events) {
+                final reward =
+                    event.rewardIds.length == 0 ? null : event.rewardIds[0];
+                final tracker = trackerModel.trackerByEventId(event.id);
+                final format = DateFormat('yyyy-MM-dd');
+                final chal = event.challengeIds.length == 0
+                    ? null
+                    : challengeModel.getChallengeById(event.challengeIds[0]);
+                final complete = tracker?.prevChallengeIds.length ==
+                    event.challengeIds.length;
+                final timeTillExpire =
```

Comment:
Given that these variables are constantly defined due to the for loop should they be of type final?